paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt

```diff
@@ -65,76 +75,82 @@ class Pagination<Item>(
 
         mNextPageLoading.postValue(true)
 
-        launch {
-            listMutex.lock()
-            try {
-                // load next page items
-                val items = dataSource.loadPage(currentList)
-                // remove already exist items
-                val newItems = items.filter { item ->
-                    val existsItem = currentList.firstOrNull { comparator.compare(item, it) == 0 }
-                    existsItem == null
-                }
-                // append new items to current list
-                val newList = newItems.plus(currentList)
-                // mark end of list if no new items
-                if (newItems.isEmpty()) {
-                    mEndOfList.value = true
-                } else {
-                    // save
-                    mStateStorage.value = newList.asState()
-                }
-                // flag
-                mNextPageLoading.value = false
-                // notify
-                nextPageListener(Result.success(newList))
-            } catch (error: Throwable) {
-                // flag
-                mNextPageLoading.value = false
-                // notify
-                nextPageListener(Result.failure(error))
+        listMutex.lock()
```

Comment:
it also should be before we change any values

Reply:
Mutex lock was moved.